qiskit/transpiler/passes/scheduling/padding/context_aware_dynamical_decoupling.py

```diff
@@ -0,0 +1,756 @@
+
+        # 3) Split the blocks into multi-qubit delay layers
+        merged_delays = []
+        for block in adjacent_delay_blocks:
+            merged_delays_in_block = self._split_adjacent_block(block, qubit_map)
+            merged_delays += merged_delays_in_block
+
+        # 4) For each multi-qubit delay, find a coloring (that takes into account neighboring
+        # CX or ECR gates) and get the DD sequence
+        all_delays = set()  # keep a list of all delays to iterate over them easily later
+
+        for merged_delay in merged_delays:
+            all_delays.update(merged_delay.ops)
+
+            # get coloring inside the n-qubit delay
+            coloring = self._get_wire_coloring(dag, merged_delay)
+
+            # get the DD sequence on the qubit
+            duration = merged_delay.end - merged_delay.start
+            checked_durations_cache = set()
+            for op in merged_delay.ops:
+                dd_circuit, start_times = self._get_dd_sequence(
+                    coloring[op.index],
+                    op.index,
+                    duration,
+                    checked_durations_cache,
+                )
+                op.replacement.compose(dd_circuit, inplace=True, copy=False)
+                if len(op.start_times) == 0:
+                    op.start_times += start_times
+                else:
+                    start_times = [op.start_times[-1] + time for time in start_times]
+
+                op.start_times += start_times
+
+        # 5) Replace each delay operation with it's individual DD sequence
+        qubit_map = dict(enumerate(dag.qubits))
+        for delay in all_delays:
+            # replace it with it's stored replacement
+            as_dag = circuit_to_dag(delay.replacement)
+            node_ids = [node._node_id for node in as_dag.topological_op_nodes()]
+            id_map = dag.substitute_node_with_dag(
+                delay.op, as_dag, {as_dag.qubits[0]: qubit_map[delay.index]}
+            )
+
+            # update node start times
+            for node_id, start_time in zip(node_ids, delay.start_times):
+                self.property_set["node_start_time"][id_map[node_id]] = start_time
+
+        return dag
+
+    @staticmethod
+    def get_orthogonal_sequence(order: int) -> tuple[list[float], list[Gate]]:
+        """Return a DD sequence of given order, where different orders are orthogonal."""
+        if order == 0:
+            spacing = [1 / 2, 1 / 2, 0]
+        elif order == 1:
+            spacing = [1 / 4, 1 / 2, 1 / 4]
+        elif order == 2:
+            spacing = [1 / 4, 1 / 4, 1 / 4, 1 / 4, 0]
+        elif order == 3:
+            spacing = [1 / 8, 1 / 4, 1 / 4, 1 / 4, 1 / 8]
+        elif order == 4:
+            spacing = [1 / 8, 1 / 4, 1 / 8, 1 / 8, 1 / 4, 1 / 8, 0]
+        elif order == 5:
+            spacing = [1 / 8, 1 / 8, 1 / 8, 1 / 4, 1 / 8, 1 / 8, 1 / 8]
+        else:
+            max_order = ContextAwareDynamicalDecoupling.MAX_ORDER
```

Comment:
Because I got to lazy to write the code that generalizes this and thought there's not frequently a device which has a qubit with more than 5 connections 😉 I'll add the code that does this generally

Reply:
Added in 13802fc36. This can now compute Walsh Hadamard sequences up to arbitrary orders.